cpp/src/strings/convert/convert_urls.cu

```diff
@@ -184,14 +184,14 @@ struct url_decoder_fn
     char* d_chars{};
 
     // utility to convert a hex char into a single byte
-    __device__ u_char hex_char_to_byte( char ch )
+    __device__ char hex_char_to_byte( char ch )
     {
         if( ch >= '0' && ch <= '9' )
-            return (ch-48);
-        if( ch >= 'A' && ch <= 'Z' )
-            return (ch-55);
-        if( ch >='a' && ch <= 'z' )
-            return (ch-87);
+            return (ch-'0');
+        if( ch >= 'A' && ch <= 'F' )
```

Comment:
Oh wow I didn't even notice the bug that was here (F vs. Z). Does this mean we need a test that uses invalid Hex characters (e.g. `G`) to make sure the results are as expected?